docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,531 @@
+    fee: (classicFeeNum + minResourceFeeNum).toString(),
+    // apply the pre-built Soroban Tx Data from simulation onto the Tx
+    sorobanData: success.transactionData.build(),
+    networkPassphrase: raw.networkPassphrase
+  });
+
+  switch (raw.operations[0].type) {
+    case 'invokeHostFunction':
+      // In this case, we don't want to clone the operation, so we drop it.
+      txnBuilder.clearOperations();
+
+      const invokeOp: Operation.InvokeHostFunction = raw.operations[0];
+      const existingAuth = invokeOp.auth ?? [];
+      txnBuilder.addOperation(
+        Operation.invokeHostFunction({
+          source: invokeOp.source,
+          func: invokeOp.func,
+          // if auth entries are already present, we consider this "advanced
+          // usage" and disregard ALL auth entries from the simulation
+          //
+          // the intuition is "if auth exists, this tx has probably been
+          // simulated before"
+          auth: existingAuth.length > 0 ? existingAuth : success.result!.auth
+        })
+      );
+      break;
+  }
+
+  return txnBuilder;
+}
+
+function isSorobanTransaction(tx: Transaction): boolean {
+  if (tx.operations.length !== 1) {
+    return false;
+  }
+
+  switch (tx.operations[0].type) {
+    case 'invokeHostFunction':
+    case 'extendFootprintTtl':
+    case 'restoreFootprint':
+      return true;
+
+    default:
+      return false;
+  }
+}
+```
+
+## Handling Archived Ledger Entries
+
+When a ledger entry is archived, it needs to be restored before the transaction can be submitted. This is indicated in the `restorePreamble` field of the result.
+
+```typescript
+interface RestorePreamble {
+  minResourceFee: string; // Recommended minimum resource fee to add when submitting the RestoreFootprint operation.
+  transactionData: string; // The recommended Soroban Transaction Data to use when submitting the RestoreFootprint operation.
+}
+```
+
+Here is an example for handling restoration using the `restorePreamble` to restore archived data:
+
+```typescript
+/**
+ * Simulates a restoration transaction to determine if restoration is needed.
+ * This function first checks the ledger entry for the given WASM hash. If the entry is found and has expired,
+ * it attempts a restoration. If the entry hasn't expired yet but the TTL needs extension, it proceeds with TTL extension.
+ * @param wasmHash - The hash of the WASM to check.
+ * @param txParams - The transaction parameters including account and signer.
+ * @returns A promise that resolves to a simulation response, indicating whether restoration or TTL extension is needed.
+ */
+export async function simulateRestorationIfNeeded(
+  wasmHash: Buffer,
+  txParams: TxParams,
+): Promise<
+  SorobanRpc.Api.SimulateTransactionRestoreResponse | string | undefined
+> {
+  try {
+    const account = await config.rpc.getAccount(txParams.account.accountId());
+    const ledgerKey = getLedgerKeyWasmId(wasmHash);
```

Comment:
where is `getLedgerKeyWasmId` defined?

Reply:
I changed this to take a contractID as the example because that code would be unrelated, and i can use the sdk's Contract class to build a footprint to get that ledger key i believe.